ci-operator/step-registry/gather/must-gather/gather-must-gather-commands.sh

```diff
@@ -4,6 +4,10 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
+export HOME=/tmp
+export WORKSPACE=${WORKSPACE:-/tmp}
```

Comment:
Do you want this to be something generically available to all steps?

Reply:
I think that would be useful imo. making sure we have the workspace and path setup like this.

Reply:
Can we use `TMPDIR`, which is [in POSIX][1]?

[1]: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03

Reply:
We can also lean on openshift/ci-tools#854 and define variables for these (example in #9676) to keep them out of the script.

Reply:
> Can we use TMPDIR, which is in POSIX?

i like WORKSPACE name more beacuse it is more clearer for it's responsibility. and works better when using it in jumped formed.

Reply:
> We can also lean on openshift/ci-tools#854 and define variables for these (example in #9676) to keep them out of the script.

The step env variables are user configurable inputs to the steps, so using it for something that is more about core function, i.e. where do i create my files seems like a mis use. I would never want someone to change the location of the workspace in a workflow because it is proabably going to break the step as we don't create or check permissions for that dir it is given to us.

Reply:
> i like WORKSPACE name more beacuse it is more clearer for it's responsibility. and works better when using it in jumped formed.

Is #9553 actually setting `WORKSPACE` anywhere?

> The step env variables are user configurable inputs to the steps...

And they're also a way to set convenient defaults, for things where the script has no opinion.

>  I would never want someone to change the location of the workspace in a workflow because it is proabably going to break the step as we don't create or check permissions for that dir it is given to us.

Then just hard-code the value, and skip the variable?  If this is something where you have some consumers that want a knob (e.g. some #9553 logic I'm missing about running these jobs with a different `WORKSPACE` on a bastion), then add an environment knob.  And if callers point it at a non-existent directory or whatever, and the job breaks as a result, that's the caller's problem.  I don't think we need defensive guards for things like that, and I don't think the lack of defensive guards is a reason to avoid a declared `env` variable in these internal tools.

Reply:
I think exposing the env from steps that are not actually user-configurable or changing them just breaks is bad API. So I do not think workspace is a good idea.

Also #9553 only uses the bash script itself and not the step, it copies the bash script to the bastion and then runs it using a unique workspace. I do not need these to available as an API on the steps level just to these scripts for reusing.